src/Config/Processor/BuilderProcessor.php

```diff
@@ -47,19 +47,42 @@ final class BuilderProcessor implements ProcessorInterface
      */
     public static function process(array $configs): array
     {
+        $addedTypes = [];
+        // map: "type name" => "provided by" for better DX, while debugging accidental type overrides in builders
+        $reservedTypesMap = \array_combine(
+            \array_keys($configs),
+            \array_fill(0, \count($configs), 'configs')
+        );
+
         foreach ($configs as &$config) {
             if (isset($config['config']['builders']) && \is_array($config['config']['builders'])) {
-                $buildersFields = self::processFieldsBuilders($config['config']['builders']);
-                $config['config']['fields'] = isset($config['config']['fields']) ? \array_merge($buildersFields, $config['config']['fields']) : $buildersFields;
+                ['fields' => $buildersFields, 'types' => $buildersTypes] = self::processFieldsBuilders(
+                    $config['config']['builders'],
+                    $reservedTypesMap
```

Comment:
Scrutinizer
/ Inspection => It seems like ``$reservedTypesMap`` can also be of type ``false``; however, parameter ``$reservedTypesMap`` of ``Overblog\GraphQLBundle\Config\Processor\BuilderProcessor::processFieldsBuilders()`` does only seem to accept ``array``, maybe add an additional type check?

Reply:
This comes from the `array_combine ` function above. It returns false when `keys` and `values `args do not have same amount of elements. Its in the `::process` method right in the beginning. 

In this case the call to array_combine does not have any inputs which could make this happen. So the problem is: When I make a check and throw, the throw at this moment can not be tested and covered and would result in coverage drop. I also can not default it to empty array or similar, as it would allow to accidentally override types.

I had the same issue at a different place, where I had thrown a LogicException which could only be triggered by chenging const values in private code and could not be triggered in unit test to be covered & tested.

Is there a way to annotate skip coverage line from counting, when code is actually unreachable? Or do you want unreachable leaves to be added at all?

Reply:
not a big deal, if this is a case not possible when using the bundle no need to fix it.